lib/tasks/spotlight.rake

```diff
@@ -4,9 +4,19 @@ namespace :spotlight do
   namespace :riiif do
     desc 'Clear the riiif cache if it exceeds 50GB'
     task clear_cache: :environment do
-      # Get the size of the tmp/network_files directory
+      # Abort if the tmp/network_files directory does not exist
+      unless Dir.exist?('tmp/network_files')
+        Rails.logger.error("Aborting spotlight:riiif:clear_cache rake task. Directory tmp/network_files does not exist.")
+        next
+      end
+      # Abort if the size of the tmp/network_files directory cannot be determined
       size = `du -sh tmp/network_files`.split("\t").first
-
+      unless $CHILD_STATUS.exitstatus.zero?
+        Rails.logger.error("Aborting spotlight:riiif:clear_cache rake task. Failed to get size of tmp/network_files.")
+        next
+      end
+      Rails.logger.info("Beginning spotlight:riiif:clear_cache rake task. Size of tmp/network_files/ is currently #{size}.")
+      Rails.logger.info("Cache exceeds 50GB threshold, starting cleanup.") if size.match(/^(\d+(?:\.\d+)?)G$/) && Regexp.last_match(1).to_f > 50
```

Comment:
Looks like this would redefine the same size check we have just below. Maybe instead move this log into the loop below w/ conditional check on `deleted_file_count` like: `Rails.logger.info("Cache exceeds 50GB threshold, starting cleanup.") if deleted_file_count = 0`